src/osd/PG.cc

```diff
@@ -3374,7 +3374,10 @@ void PG::update_snap_map(
 	int r = snap_mapper.remove_oid(
 	  i->soid,
 	  &_t);
-	assert(r == 0);
+	if (!(r == 0 || r == -ENOENT)) {
```

Comment:
@gmayyyha I do believe with @liewegas that there are likely less chances of '  if (got.empty())    return -ENOENT;` is failing to get keys.

Reply:
@amitkumar50, @liewegas I see several other methods also use this judgment, only as a workaround?
https://github.com/ceph/ceph/blob/f71af8125129ebaa8a02712ca9d6e45fb9a4cd12/src/osd/PG.cc#L490-L498